pallets/living-assets-ownership/src/functions.rs

```diff
@@ -1,22 +1,54 @@
 //! Contains helper and utility functions of the pallet
 use super::*;
-use frame_support::{ensure, sp_runtime::DispatchResult};
+use frame_support::sp_runtime::{
+	traits::{CheckedAdd, One},
+	DispatchResult,
+};
 
 impl<T: Config> Pallet<T> {
-	/// See [Self::create_collection]
-	pub fn do_create_collection(
-		collection_id: T::CollectionId,
-		who: T::AccountId,
-	) -> DispatchResult {
-		ensure!(
-			!OwnerOfCollection::<T>::contains_key(collection_id),
-			Error::<T>::CollectionAlreadyExists
-		);
+	/// Attempts to create a new collection
+	///
+	/// This function is intended to be used as the implementation for the
+	/// [Self::create_collection] public interface. It first retrieves the current
+	/// collection count to use as the new collection's ID. It then inserts a new
+	/// entry into the `OwnerOfCollection` map, mapping the new collection's ID to
+	/// the owner's account ID.
+	///
+	/// After this, the function attempts to increment the collection counter by 1.
+	/// If this operation would result in an overflow, the function returns early
+	/// with an [Error::CollectionIdOverflow].
+	///
+	/// Finally, if all operations were successful, the function emits a
+	/// [Event::CollectionCreated] event and returns `Ok(())`.
+	///
+	/// # Arguments
+	///
+	/// * `who` - The account ID of the new collection's owner.
+	///
+	/// # Return
+	///
+	/// Returns a [DispatchResult] indicating the outcome of the operation. If the
+	/// operation was successful, the function returns `Ok(())`. If the operation
+	/// was not successful, the function returns `Err(e)`, where `e` is the error
+	/// that occurred.
+	pub fn do_create_collection(who: T::AccountId) -> DispatchResult {
+		// Retrieve the current collection count to use as the new collection's ID
+		let collection_id = Self::collection_counter();
 
+		// Insert a new entry into the OwnerOfCollection map, mapping the new
+		// collection's ID to the owner's account ID
 		OwnerOfCollection::<T>::insert(collection_id, &who);
 
+		// Attempt to increment the collection counter by 1. If this operation
+		// would result in an overflow, return early with an error
+		let counter =
```

Comment:
would suggest [this](https://docs.rs/sp-runtime/latest/sp_runtime/traits/trait.Saturating.html#method.saturating_inc) instead

Reply:
https://substrate.stackexchange.com/questions/9441/understanding-the-use-case-of-saturating-inc

----
saturation_inc does not allow to recognize an error case as the overflow I am addressing